r_sort.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+from merge_sort import timed_func
+
+
+@timed_func
+def r_sort(sequence):
+    digit = 1
+    while True:
+        bins = [[] for i in range(10)]
+        for x in sequence:
+            # the expression in bins returns the number in digit place
+            bins[x % 10 ** digit // 10 ** (digit -1)].append(x)
+        sequence = [number for x in bins for number in x]
+        digit += 1
+        if len(sequence) == len(bins[0]):
+            return sequence
+
+
+def str_index(_str, digit):
+    """Returns the ord of the text at the index of digit"""
+    try:
+        index = ord(_str[digit])
+        return index + 1
+    except IndexError:
+        return 0
+
+
+@timed_func
+def r_sort_alpha(sequence):
+    max_digit = 0
+    # find the length of longest string
+    for _str in sequence:
+        if len(_str) > max_digit:
+            max_digit = len(_str)
+    for digit in range(max_digit + 1, -1, -1):
```

Comment:
good!  I'd suggest using xrange instead, but nice job remembering the step argument!